src/Npgsql/Replication/Internal/LogicalReplicationConnectionExtensions.cs

```diff
@@ -79,21 +79,44 @@ Task<ReplicationSlotOptions> CreateLogicalReplicationSlotCore()
             if (isTemporary)
                 builder.Append(" TEMPORARY");
             builder.Append(" LOGICAL ").Append(outputPlugin);
-            builder.Append(slotSnapshotInitMode switch
+            if (connection.PostgreSqlVersion.Major >= 15 && (slotSnapshotInitMode.HasValue || twoPhase))
             {
-                // EXPORT_SNAPSHOT is the default since it has been introduced.
-                // We don't set it unless it is explicitly requested so that older backends can digest the query too.
-                null => string.Empty,
-                LogicalSlotSnapshotInitMode.Export => " EXPORT_SNAPSHOT",
-                LogicalSlotSnapshotInitMode.Use => " USE_SNAPSHOT",
-                LogicalSlotSnapshotInitMode.NoExport => " NOEXPORT_SNAPSHOT",
-                _ => throw new ArgumentOutOfRangeException(nameof(slotSnapshotInitMode),
-                    slotSnapshotInitMode,
-                    $"Unexpected value {slotSnapshotInitMode} for argument {nameof(slotSnapshotInitMode)}.")
-            });
-            if (twoPhase)
-                builder.Append(" TWO_PHASE");
-
+                builder.Append('(');
```

Comment:
Is a leading space needed here (between outputPlugin and the opening parenthesis)?

Reply:
IDK.
I don't think so but I'll test it.